core/src/main/resources/lib/form/number.jelly

```diff
@@ -76,12 +76,25 @@ THE SOFTWARE.
   <j:set var="name" value="${attrs.name ?: '_.'+attrs.field}"/>
   <j:set var="default" value="${attrs.default ?: ''}"/>
   <j:set var="value" value="${attrs.value ?: instance[attrs.field] ?: default}"/>
-  <m:input xmlns:m="jelly:hudson.util.jelly.MorphTagLibrary"
-         class="setting-input ${attrs.checkUrl!=null?'validated':''} ${attrs.clazz}"
-         name="${name}"
-         value="${value}"
-         type="number"
-         ATTRIBUTES="${attrs}" EXCEPT="field clazz" />
+
+  <j:choose>
+    <j:when test="${displayOnlyMode}">
+      <j:choose>
+        <j:when test="${empty(value)}">
+          <span class="jenkins-not-applicable">N/A</span>
```

Comment:
Would it make sense to extract this `<j:choose>` to its own jelly template?

Reply:
Will take a look at this later on, fixed the other comments up

Reply:
Done, in all except password as it has a different default text rendering handling